pulsar-client-cpp/lib/HTTPLookupService.cc

```diff
@@ -148,132 +148,145 @@ void HTTPLookupService::handleNamespaceTopicsHTTPRequest(NamespaceTopicsPromise
-            curl_easy_setopt(handle, CURLOPT_SSL_VERIFYPEER, 0L);
-        } else {
-            curl_easy_setopt(handle, CURLOPT_SSL_VERIFYPEER, 1L);
-        }
+            if (tlsAllowInsecure_) {
+                curl_easy_setopt(handle, CURLOPT_SSL_VERIFYPEER, 0L);
+            } else {
+                curl_easy_setopt(handle, CURLOPT_SSL_VERIFYPEER, 1L);
+            }
 
-        if (!tlsTrustCertsFilePath_.empty()) {
-            curl_easy_setopt(handle, CURLOPT_CAINFO, tlsTrustCertsFilePath_.c_str());
-        }
+            if (!tlsTrustCertsFilePath_.empty()) {
+                curl_easy_setopt(handle, CURLOPT_CAINFO, tlsTrustCertsFilePath_.c_str());
+            }
 
-        curl_easy_setopt(handle, CURLOPT_SSL_VERIFYHOST, tlsValidateHostname_ ? 1L : 0L);
+            curl_easy_setopt(handle, CURLOPT_SSL_VERIFYHOST, tlsValidateHostname_ ? 1L : 0L);
 
-        if (authDataContent->hasDataForTls()) {
-            curl_easy_setopt(handle, CURLOPT_SSLCERT, authDataContent->getTlsCertificates().c_str());
-            curl_easy_setopt(handle, CURLOPT_SSLKEY, authDataContent->getTlsPrivateKey().c_str());
+            if (authDataContent->hasDataForTls()) {
+                curl_easy_setopt(handle, CURLOPT_SSLCERT, authDataContent->getTlsCertificates().c_str());
+                curl_easy_setopt(handle, CURLOPT_SSLKEY, authDataContent->getTlsPrivateKey().c_str());
+            }
         }
-    }
-
-    LOG_INFO("Curl Lookup Request sent for " << completeUrl);
-
-    // Make get call to server
-    res = curl_easy_perform(handle);
-
-    // Free header list
-    curl_slist_free_all(list);
 
-    Result retResult = ResultOk;
-
-    switch (res) {
-        case CURLE_OK:
-            long response_code;
-            curl_easy_getinfo(handle, CURLINFO_RESPONSE_CODE, &response_code);
-            LOG_INFO("Response received for url " << completeUrl << " code " << response_code);
-            if (response_code == 200) {
-                retResult = ResultOk;
-            } else {
+        LOG_INFO("Curl [" << reqCount << "] Lookup Request sent for " << completeUrl);
+
+        // Make get call to server
+        res = curl_easy_perform(handle);
+
+        long response_code;
+        curl_easy_getinfo(handle, CURLINFO_RESPONSE_CODE, &response_code);
+        LOG_INFO("Response received for url " << completeUrl << " response_code " << response_code
+                                              << " curl res " << res);
+
+        // Free header list
+        curl_slist_free_all(list);
+
+        switch (res) {
+            case CURLE_OK:
+                long response_code;
+                curl_easy_getinfo(handle, CURLINFO_RESPONSE_CODE, &response_code);
+                LOG_INFO("Response received for url " << completeUrl << " code " << response_code);
+                if (response_code == 200) {
+                    retResult = ResultOk;
+                } else if (response_code == 307 || response_code == 302 || response_code == 301) {
```

Comment:
It's better to add a common function to check if the response code needs redirection because the check is used in two places.

```c++
inline bool needRedirection(long code) {
    return code % 300 < 10;  // Here I just assume other 30x code also needs redirection
}
```